torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -2239,4 +2239,8 @@ void Reducer::remove_autograd_hooks() {
   hooks_.clear();
 }
 
+bool Reducer::has_finalized() {
+  return !require_finalize_;
```

Comment:
I think we have a lock around this access, do we need that here?

Reply:
Good catch!

Reply:
Need to add the lock?

Reply:
Ah I thought I added it in, will fix it